common/src/reserved.rs

```diff
@@ -507,6 +513,43 @@ mod tests {
         );
     }
 
+    #[test]
+    fn test_apply_delegate_on_consensus_failure() {
+        // given
+        setup_test();
+        let (mut state, keys) = generate_standard_genesis(3);
+
+        // delegator: member-0000, delegatee: member-0002
+        let delegator = state.members[0].clone();
+        let delegator_private_key = keys[0].1.clone();
+
+        // when
+        let data: DelegationTransactionData = DelegationTransactionData {
+            delegator: delegator.name.clone(),
+            delegatee: delegator.name.clone(),
+            governance: false,
+            block_height: 0,
+            timestamp: 0,
+            chain_name: state.genesis_info.chain_name.clone(),
+        };
+        let proof = TypedSignature::sign(&data, &delegator_private_key).unwrap();
+
+        let tx = TxDelegate { data, proof };
+
+        let new_state = state.apply_delegate(&tx);
+
+        // then
+        //
+        if let Err(s) = new_state {
+            assert_eq!(
+                s,
+                format!("delegator and delegatee are the same: {}", delegator.name)
+            );
+        } else {
+            panic!("Unexpected error");
```

Comment:
Try more helpful error message.
How about `it must fail when the delegator and the delegatee are the same`